### PR TITLE
[WIP] A tensor parallel API for beginners

### PR DESCRIPTION
**Pull request by siddharth9820**
- Attempting to create a tensor parallel API which requires minimal changes to the model definition.
- Using the popular nanoGPT benchmark as a testbed - https://github.com/axonn-ai/nanoGPT/pull/1

### CODE REVIEW

**Comment by siddharth9820** [comment]
To Dos - 

- [x] Add grad normalization support in axonn/intra_layer 
- [x] Make nn.kaiming_init the default in axonn/intra_layer/fully_connected
- [x] Change Tensor_Parallel_Linear to Linear

File: axonn/intra_layer/gradient_normalization.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import torch
+from torch._six import inf
```
Comment (siddharth9820) [inline axonn/intra_layer/gradient_normalization.py]: @jwendlan pointed out that this doesnt work with pytorch 2.0

**Comment by siddharth9820** [comment]
Added CI tests for -
1. Easy tensor parallelism
2. Gradient clipping
3. bias True and False
